common/types/ethasset.go

```diff
@@ -16,6 +19,26 @@ func (asset EthAsset) String() string {
 	return ethcommon.Address(asset).Hex()
 }
 
+// MarshalText returns the hex representation of the EthAsset
+func (asset EthAsset) MarshalText() ([]byte, error) {
+	return []byte(asset.String()), nil
+}
+
+// UnmarshalText assigns the EthAsset from the input text
+func (asset *EthAsset) UnmarshalText(input []byte) error {
+	inputStr := string(input)
+	switch {
+	case strings.EqualFold(inputStr, "ETH"):
+		*asset = EthAsset{}
+		return nil
+	case ethcommon.IsHexAddress(inputStr):
+		*asset = EthAsset(ethcommon.HexToAddress(inputStr))
+		return nil
+	default:
+		return fmt.Errorf("invalid asset value %q", inputStr)
+	}
+}
+
```

Comment:
I learned something in this PR:
```
type EthAsset ethcommon.Address     // does not inherit methods (declaration higher up in this file)
type EthAsset = ethcommon.Address   // has all methods available on ethcomon.Address
```
The go-ethereum project has these methods for its Address type, but we are doing special handling of some asset names (only "ETH" for now), so we couldn't have used their methods anyways. The existing declaration type is ideal. With our Hash type, we are doing assignment in the type declaration.